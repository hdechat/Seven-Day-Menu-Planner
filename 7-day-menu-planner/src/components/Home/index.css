.select-category {
  border: solid 1px white;
  margin: 1rem auto;
  width: 9rem;
}

.find-your-recipe {
  margin-bottom: 0;
}

.english-spanish {
  display: flex;
  justify-content: space-around;
  margin: auto 27rem;
}

.loading-gif {
  margin-top: 2rem;
  width: 40%;
}

.error-message {
  color: black;
  font-size: 30px;
  margin: 0;
}

.search {
  font-size: 16px;
  height: 2rem;
  border-radius: 0%;
  width: 9.2rem;;
}

.search::placeholder {
  font-size: 16px;
}

.display-cards {
  height: 70vh;
  margin: auto;
  overflow-y: scroll;
  width: 90%;
}

.filters {
  border: solid grey thin;
  background: lightblue;
}

span {
  color: yellowgreen;
  font-size: 1rem
}

select#category {
  background: transparent;
  background-image: none;
  border: none;
  color: white;
  font-size: 12px;
  height: 2rem;
  outline: none;
  padding: 5px 8px;
}

select:hover {
  color: black;
  font-size: .8rem;
}

@media screen and (max-width: 350px) {
  .find-your-recipe {
    display: none;
  }

  .english, .spanish {
    margin-left: 2rem;
    margin-top: 5rem;
  }

  select#category {
    font-size: 20px;
    height: 3rem;
  }

  .filters {
    width: 14.1rem;
    margin-left: 1rem;
  }
}
