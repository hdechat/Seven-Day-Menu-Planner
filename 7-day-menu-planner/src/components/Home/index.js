import React, { Component } from 'react';
import RecipeCardsDisplay from '../RecipeCardsDisplay';
import './index.css';

class Home extends Component {
  constructor(props) {
    super(props);

    this.state = { 
      userSearch: '',
      spanishSearch: ''
    };
  }

  render() {
    const { 
      recipeCards, 
      chooseCategory, 
      recipesIsLoading, 
      recipesHasErrored } = this.props;

    const displayLoadingMessage = () => {
      return recipesIsLoading ?
        <p className="loading-message">Your Recipes Are Loading</p> : null;
    };

    const displayErrorMessage = () => {
      return recipesHasErrored ?
        <p className="error-message">{recipesHasErrored}</p> : null;
    };

    const displayWelcomeMessage = () => {
      return !recipeCards.length ?
        <p className="welcome-message">Welcome to your eazy-peazy seven-day meal planner!</p> : null;
    }

    return (
      <section className="home">
        <h2>Find Your Recipe / Encuentra Tu Receta<span>BETA</span></h2>
        <div className="english-spanish">
          <div className="english">  
            <div className="select-category">
              <select 
                onChange={event => chooseCategory('api', event.target.value)}
                id='category'>
                <option>SELECT CATEGORY</option>
                <option value='paleo'>Paleo</option>
                <option value='vegetarian'>Vegetarian</option>
                <option value='vegan'>Vegan</option>
              </select>
            </div>
            <input 
              className='search' 
              type='search'
              placeholder='Search'
              value={this.state.userSearch}
              onChange={event => this.setState({userSearch: event.target.value})}
              onKeyPress={e => e.key === 'Enter' ? chooseCategory('api', this.state.userSearch) : ''} />
          </div>
          <div className="spanish">
            <div className="select-category">
              <select 
                onChange={event => chooseCategory('test-es', event.target.value)}
                id='category'>
                <option>CATEGORIAS</option>
                <option value='paleo'>Paleo</option>
                <option value='vegetarian'>Vegetariano</option>
                <option value='vegan'>Vegano</option>
              </select>
            </div>
            <input 
              className='search' 
              type='search'
              placeholder='Buscar'
              value={this.state.spanishSearch}
              onChange={event => this.setState({spanishSearch: event.target.value})}
              onKeyPress={e => e.key === 'Enter' ? chooseCategory('test-es', this.state.spanishSearch) : ''} />
          </div>
        </div>
          {displayLoadingMessage()}
          {displayErrorMessage()}
          {displayWelcomeMessage()}
        <div className="display-cards">
          <RecipeCardsDisplay recipeCards={recipeCards} />
        </div>
      </section>
    );
  }
}

export default Home;
