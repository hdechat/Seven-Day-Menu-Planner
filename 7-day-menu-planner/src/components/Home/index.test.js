import React from 'react';
import Home from './index.js';
import { shallow } from 'enzyme';

describe('Home', () => {
  let wrapper;
  let mockFunction;

  beforeEach(() => {
    mockFunction = jest.fn();

    wrapper = shallow(<Home 
      recipesHasErrored={''}
       recipesIsLoading={false} 
       recipeCards={[]} 
       chooseCategory={mockFunction} />);
  });

  it('matches snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('should call chooseCategory with the correct params when category selected', () => {
    wrapper.find('select').simulate('change', {target: {value: 'paleo'}});

    expect(mockFunction).toHaveBeenCalledWith('paleo');
  });

  it('should render message if recipesIsLoading is true', () => {
    wrapper.setProps({recipesIsLoading: true});

    expect(wrapper.find('.loading-message').length).toEqual(1);
  });

  it('should render error if fetch has errored', () => {
    wrapper.setProps({recipesHasErrored: 'Something went wrong.'});
    expect(wrapper.find('.error-message').length).toEqual(1);
  });

  it('should display welcome message if no recipe cards are rendered', () => {
    expect(wrapper.find('.welcome-message').length).toEqual(1);
    wrapper.setProps({ recipeCards: [{title: 'recipe'}] });

    expect(wrapper.find('.welcome-message').length).toEqual(0);
  });

  it('should set state with onChange', () => {
    wrapper.find('.search').simulate('change', {target: {value: 'high-protein'}});

    expect(wrapper.state('userSearch')).toEqual('high-protein');
  });

  it('should call chooseCategory when user enters search term', () => {
    wrapper.setState({ userSearch: 'high-protein' })
    wrapper.find('.search').simulate('keypress', {key: 'Enter'});

    expect(mockFunction).toHaveBeenCalledWith('high-protein');
  });
});
