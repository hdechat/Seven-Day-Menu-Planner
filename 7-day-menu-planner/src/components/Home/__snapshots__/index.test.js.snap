// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home matches snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Home
    chooseCategory={[Function]}
    recipeCards={Array []}
    recipesHasErrored=""
    recipesIsLoading={false}
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <h2>
          Select Category
</h2>,
        <div
          className="select-category"
>
          <select
                    id="category"
                    onChange={[Function]}
          >
                    <option>
                              SELECT CATEGORY
                    </option>
                    <option
                              value="paleo"
                    >
                              Paleo
                    </option>
                    <option
                              value="vegetarian"
                    >
                              Vegetarian
                    </option>
                    <option
                              value="vegan"
                    >
                              Vegan
                    </option>
          </select>
</div>,
        <input
          className="search"
          placeholder="Search"
          type="text"
/>,
        null,
        null,
        <p
          className="welcome-message"
>
          Welcome to your easy-peasy seven day meal planner!
</p>,
        <div
          className="display-cards"
>
          <RecipeCardsDisplay
                    recipeCards={Array []}
          />
</div>,
      ],
      "className": "home",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Select Category",
        },
        "ref": null,
        "rendered": "Select Category",
        "type": "h2",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <select
            id="category"
            onChange={[Function]}
>
            <option>
                        SELECT CATEGORY
            </option>
            <option
                        value="paleo"
            >
                        Paleo
            </option>
            <option
                        value="vegetarian"
            >
                        Vegetarian
            </option>
            <option
                        value="vegan"
            >
                        Vegan
            </option>
</select>,
          "className": "select-category",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <option>
                SELECT CATEGORY
</option>,
              <option
                value="paleo"
>
                Paleo
</option>,
              <option
                value="vegetarian"
>
                Vegetarian
</option>,
              <option
                value="vegan"
>
                Vegan
</option>,
            ],
            "id": "category",
            "onChange": [Function],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "SELECT CATEGORY",
              },
              "ref": null,
              "rendered": "SELECT CATEGORY",
              "type": "option",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Paleo",
                "value": "paleo",
              },
              "ref": null,
              "rendered": "Paleo",
              "type": "option",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Vegetarian",
                "value": "vegetarian",
              },
              "ref": null,
              "rendered": "Vegetarian",
              "type": "option",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Vegan",
                "value": "vegan",
              },
              "ref": null,
              "rendered": "Vegan",
              "type": "option",
            },
          ],
          "type": "select",
        },
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "className": "search",
          "placeholder": "Search",
          "type": "text",
        },
        "ref": null,
        "rendered": null,
        "type": "input",
      },
      null,
      null,
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Welcome to your easy-peasy seven day meal planner!",
          "className": "welcome-message",
        },
        "ref": null,
        "rendered": "Welcome to your easy-peasy seven day meal planner!",
        "type": "p",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <RecipeCardsDisplay
            recipeCards={Array []}
/>,
          "className": "display-cards",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "recipeCards": Array [],
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        "type": "div",
      },
    ],
    "type": "section",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <h2>
            Select Category
</h2>,
          <div
            className="select-category"
>
            <select
                        id="category"
                        onChange={[Function]}
            >
                        <option>
                                    SELECT CATEGORY
                        </option>
                        <option
                                    value="paleo"
                        >
                                    Paleo
                        </option>
                        <option
                                    value="vegetarian"
                        >
                                    Vegetarian
                        </option>
                        <option
                                    value="vegan"
                        >
                                    Vegan
                        </option>
            </select>
</div>,
          <input
            className="search"
            placeholder="Search"
            type="text"
/>,
          null,
          null,
          <p
            className="welcome-message"
>
            Welcome to your easy-peasy seven day meal planner!
</p>,
          <div
            className="display-cards"
>
            <RecipeCardsDisplay
                        recipeCards={Array []}
            />
</div>,
        ],
        "className": "home",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Select Category",
          },
          "ref": null,
          "rendered": "Select Category",
          "type": "h2",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <select
              id="category"
              onChange={[Function]}
>
              <option>
                            SELECT CATEGORY
              </option>
              <option
                            value="paleo"
              >
                            Paleo
              </option>
              <option
                            value="vegetarian"
              >
                            Vegetarian
              </option>
              <option
                            value="vegan"
              >
                            Vegan
              </option>
</select>,
            "className": "select-category",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <option>
                  SELECT CATEGORY
</option>,
                <option
                  value="paleo"
>
                  Paleo
</option>,
                <option
                  value="vegetarian"
>
                  Vegetarian
</option>,
                <option
                  value="vegan"
>
                  Vegan
</option>,
              ],
              "id": "category",
              "onChange": [Function],
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "SELECT CATEGORY",
                },
                "ref": null,
                "rendered": "SELECT CATEGORY",
                "type": "option",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "Paleo",
                  "value": "paleo",
                },
                "ref": null,
                "rendered": "Paleo",
                "type": "option",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "Vegetarian",
                  "value": "vegetarian",
                },
                "ref": null,
                "rendered": "Vegetarian",
                "type": "option",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "Vegan",
                  "value": "vegan",
                },
                "ref": null,
                "rendered": "Vegan",
                "type": "option",
              },
            ],
            "type": "select",
          },
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "className": "search",
            "placeholder": "Search",
            "type": "text",
          },
          "ref": null,
          "rendered": null,
          "type": "input",
        },
        null,
        null,
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Welcome to your easy-peasy seven day meal planner!",
            "className": "welcome-message",
          },
          "ref": null,
          "rendered": "Welcome to your easy-peasy seven day meal planner!",
          "type": "p",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <RecipeCardsDisplay
              recipeCards={Array []}
/>,
            "className": "display-cards",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "recipeCards": Array [],
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          "type": "div",
        },
      ],
      "type": "section",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
